core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -130,4 +128,26 @@ public void init(ComponentRegistry componentRegistry) {
          ((InjectableComponent) f).inject(componentRegistry);
       }
    }
+
+   @Override
+   public final boolean isReturnValueExpected() {
+      return true;
+   }
+
+   @Override
+   public Metadata getMetadata() {
```

Comment:
Yes, and there's the concern about missing value during early rebalance: I'll create the reproducer, please be a bit patient with me; I am working on resolving the other issues you've raised & while I've forked myself recently I can't do twice as much work :)